File: TelerikUIExamplesInCSharp/Examples/ListView/ListViewAnimations.cs
using System;

using Foundation;
using UIKit;
using CoreGraphics;

using TelerikUI;

namespace Examples
{
	public class ListViewAnimations: ExampleViewController
	{
		TKDataSource dataSource = new TKDataSource();
		TKListView listView = new TKListView();

		public override void ViewDidLoad ()
		{
			AddOption ("Scale in", ScaleInSelected);
			AddOption ("Fade in", FadeInSelected);
			AddOption ("Slide in", SlideInSelected);

			base.ViewDidLoad ();

			this.dataSource.LoadDataFromJSONResource ("ListViewSampleData", "json", "photos");

			this.dataSource.Settings.ListView.CreateCell ((TKListView listView, NSIndexPath indexPath, NSObject item) => {
				return listView.DequeueReusableCell("cell", indexPath) as TKListViewCell;
			});
			this.dataSource.Settings.ListView.InitCell (delegate (TKListView listView, NSIndexPath indexPath, TKListViewCell cell, NSObject item) {
				cell.ImageView.Image = new UIImage (this.dataSource.Items[indexPath.Row] as NSString);
			});

			this.listView.Frame = this.View.Bounds;
			this.listView.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.listView.WeakDataSource = this.dataSource;
			this.listView.RegisterClassForCell (new ObjCRuntime.Class (typeof(AnimationListCell)), "cell");
			this.View.AddSubview (this.listView);

			var layout = new TKListViewGridLayout();
			if (UIDevice.CurrentDevice.UserInterfaceIdiom == UIUserInterfaceIdiom.Pad) {
				layout.SpanCount = 3;
			}
			else {
				layout.SpanCount = 2;
			}
			layout.ItemSize = new CGSize(130, 180);
			layout.ItemSpacing = 10;
			layout.LineSpacing = 10;
			layout.ItemAlignment = TKListViewItemAlignment.Center;
			layout.ItemAppearAnimation = TKListViewItemAnimation.Scale;
			layout.AnimationDuration = 0.4f;
			this.listView.Layout = layout;
		}

		void ScaleInSelected(object sender, EventArgs e)
		{
			TKListViewLinearLayout layout = (TKListViewLinearLayout)this.listView.Layout;
			layout.ItemAppearAnimation = TKListViewItemAnimation.Scale;
		}

		void FadeInSelected(object sender, EventArgs e)
		{
			TKListViewLinearLayout layout = (TKListViewLinearLayout)this.listView.Layout;
			layout.ItemAppearAnimation = TKListViewItemAnimation.Fade;
		}

		void SlideInSelected(object sender, EventArgs e)
		{
			TKListViewLinearLayout layout = (TKListViewLinearLayout)this.listView.Layout;
			layout.ItemAppearAnimation = TKListViewItemAnimation.Slide;
		}
	}
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/ListViewAnimations.swift
//
//  ListViewAnimations.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

import UIKit

class ListViewAnimations: ExampleViewController {

    let listView = TKListView()
    let dataSource = TKDataSource()
    
    override init(nibName nibNameOrNil: String!, bundle nibBundleOrNil: NSBundle!) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
        
        self.addOption("Scale in") { self.scaleInSelected() }
        self.addOption("Fade in") { self.fadeInSelected() }
        self.addOption("Slide in") { self.slideInSelected() }
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        self.dataSource.loadDataFromJSONResource("ListViewSampleData", ofType: "json", rootItemKeyPath: "photos")
        
        self.dataSource.settings.listView.createCell { (TKListView listView, NSIndexPath indexPath, AnyObject item) -> TKListViewCell! in
            return listView.dequeueReusableCellWithReuseIdentifier("cell", forIndexPath: indexPath) as! TKListViewCell
        }
        
        self.dataSource.settings.listView.initCell { (listView: TKListView!, indexPath: NSIndexPath!, cell: TKListViewCell!, item: AnyObject!) -> Void in
            cell.imageView.image = UIImage(named: self.dataSource.items[indexPath.row] as! String)
        }

        self.listView.frame = self.view.bounds
        self.listView.autoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight
        self.listView.dataSource = self.dataSource
        self.listView.registerClass(AnimationListCell.self, forCellWithReuseIdentifier: "cell")
        self.view.addSubview(self.listView)
        
        let layout = TKListViewGridLayout()
        if UIDevice.currentDevice().userInterfaceIdiom == UIUserInterfaceIdiom.Pad {
            layout.spanCount = 3
        }
        else {
            layout.spanCount = 2
        }
        layout.itemSize = CGSizeMake(130, 180)
        layout.itemSpacing = 10
        layout.lineSpacing = 10
        layout.itemAlignment = TKListViewItemAlignment.Center
        layout.itemAppearAnimation = TKListViewItemAnimation.Scale
        layout.animationDuration = 0.4
        self.listView.layout = layout
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func fadeInSelected() {
        let layout = listView.layout as! TKListViewLinearLayout
        layout.itemAppearAnimation = TKListViewItemAnimation.Fade
    }
    
    func slideInSelected() {
        let layout = listView.layout as! TKListViewLinearLayout
        layout.itemAppearAnimation = TKListViewItemAnimation.Slide
    }
    
    func scaleInSelected() {
        let layout = listView.layout as! TKListViewLinearLayout
        layout.itemAppearAnimation = TKListViewItemAnimation.Scale
    }
}
